daemon/daemon.go

```diff
@@ -490,17 +506,20 @@ func (d *Daemon) HandleRestart(t state.RestartType) {
 
 		d.mu.Lock()
 		defer d.mu.Unlock()
-		// remember we need to restart the system
+		// save the restart kind to write out a maintenance.json in a bit
 		d.restartSystem = t
 	case state.RestartSocket:
 		d.mu.Lock()
 		defer d.mu.Unlock()
+		// save the restart kind to write out a maintenance.json in a bit
+		d.restartSystem = t
```

Comment:
restartSystem is getting more and more overloaded in meaning, we might need a new name, approach

Reply:
I will think of a new name for this

Reply:
I went with `activeRestartState`, thoughts?